Taskfile.yaml

```diff
@@ -0,0 +1,133 @@
+# Taskfile for 'vugu'
+# 
+# The Taskfile builds vugu, runs all the tests and lints the source code.
+
+version: '3'
+
+# TODO: Warning this task file assumes Linux is the OS. The Taskfile will need to reworked to strip the Linux only shell cmds
+tasks:
+  # Build everything, run all the tests (using go, tinygo and tinygo in docker) and lint the codebase
+  all:
+    preconditions:
+      - task: check-cmds-exist
+    cmds:
+      - task: clean-vgform
+      - task: clean-wasm-test-suite
+      - task: build
+      - task: test
+      - task: test-wasm
+      - task: lint
+
+
+  # Build the vugu packages and the vugugen and vugufmt commands
+  build:
+    cmds:
+      # note we now use the full package names
+      # we have to run go generate in the vgform package
+      - go generate github.com/vugu/vugu/vgform
```

Comment:
I don't think we should be re-go-generating the vgform package as part of the build.  The usual process for go generate is that when you edit a package then you go generate it and then the output just stays committed to the repo (there are various examples of this in the Go standard library).  So unless there's a really good reason to include this, I think we can nuke that go generate line

Reply:
No we need this, at least of the minute.

Without PR261 this generates the "*_vgen" file that has to exist in the repo before the vgfrom package will build.
With PR262 this generates the "*_gen" in the vgform package, which likewise has to exist.

Without these files (named either way) the build fails - because the `vgform` tests fails. If you happen to have both versions on disk at the same time (as I have had when switching between the PR261 branch and this one) then the build will also fail.

If you want to remove it, then we need to somehow ensure that either the `go generate` has run, or the files are present (and correct) in the repo.

As it's more or less impossible to tell the difference between this situations the `Taskfile` takes the only safe option and regenerates everything. This way we are sure everything necessary is on disk before the `go build` steps. The build of everything in `vugu` shouldn't need any special steps to be taken by the developer before hand.